### DISCUSSION TITLE
Can we use github as storage option for apicurio registry?

### DISCUSSION THREAD

**Discussion by mdb108** [Apicurio/apicurio-registry#4871] [General]
As I am looking out some of the out of box thing so should we able to use the github as storage option to store and retrieve the artifacts?

**Comment by EricWittmann** [comment]
Currently this is not possible with the released version of Registry (2.6.x).  However, we have some experimental work for this in the latest major version (3.x) which is not yet released.

So, in the `main` branch of registry there is some work to make a Read Only storage based on Git, where Registry would consume information from a git repository.  In this mode, **all** modifications need to be made in Git.  All write operations to Registry will fail in this mode - if you try to use the REST API to make changes, you'll be rejected.

If you want to have a look at this work you can find it here:

https://github.com/Apicurio/apicurio-registry/tree/main/app/src/main/java/io/apicurio/registry/storage/impl/gitops

@jsenko is the author and can answer questions :)

**Comment by EricWittmann** [comment]
BTW: we would love to have community collaboration in this area!  In whatever form that takes:  feedback or PRs.  :)
